src/coord/cartesian/AxisModel.ts

```diff
@@ -57,6 +57,16 @@ export class CartesianAxisModel extends ComponentModel<CartesianAxisOption>
     getCoordSysModel(): GridModel {
         return this.getReferringComponents('grid', SINGLE_REFERRING).models[0] as GridModel;
     }
+
+    getNeedCrossZero(): boolean {
+        const option = this.option;
+
+        if (option.type === 'value') {
+            return !option.scale;
+        }
+
+        return !axisDefault[option.type].scale;
```

Comment:
Seems it means the option provided to developers shows it can be configurable. But actually it doesn't. 

I think it's better to remove `scale` type definition and default values in `log`, `time`, `category` axis. And check if scale is `'interval'` in https://github.com/apache/echarts/blob/master/src/coord/scaleRawExtentInfo.ts#L99